sale_margin_update/wizard/recalculate_price_margin.py

```diff
@@ -0,0 +1,42 @@
+# Copyright 2024 SDi Sidoo Soluciones S.L. <www.sidoo.es>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+
+
+class SaleRecalculatePriceMargin(models.TransientModel):
+    _name = "sale.recalculate.price.margin"
+    _description = "Recalculate Price By Margin"
+
+    sale_margin_percent = fields.Float(
+        string="Margin (%)",
+        digits="Product Price",
+        required=True,
```

Comment:
@ljimenezsidoo For a better UX, I would have added a default here with the curent line margin.